src/main/java/com/celements/cleverreach/MailingConfig.java

```diff
@@ -0,0 +1,75 @@
+package com.celements.cleverreach;
+
+import static com.google.common.base.Preconditions.*;
+import static com.google.common.base.Strings.*;
+
+import javax.annotation.concurrent.Immutable;
+import javax.validation.constraints.NotNull;
+
+@Immutable
+public class MailingConfig {
+
+  public static class Builder {
+
+    private String id;
+    private String subject;
+    private String contentHtml;
+    private String contentPlain;
+
+    public Builder setId(@NotNull String id) {
+      this.id = !isNullOrEmpty(id) ? id : null;
+      return this;
+    }
+
+    public Builder setSubject(@NotNull String subject) {
+      this.subject = !isNullOrEmpty(subject) ? subject : null;
+      return this;
+    }
+
+    public Builder setContentHtml(@NotNull String contentHtml) {
+      this.contentHtml = !isNullOrEmpty(contentHtml) ? contentHtml : null;
+      return this;
+    }
+
+    public Builder setContentPlain(@NotNull String contentPlain) {
+      this.contentPlain = !isNullOrEmpty(contentPlain) ? contentPlain : null;
+      return this;
+    }
+
+    public MailingConfig build() {
+      checkState(!isNullOrEmpty(id));
```

Comment:
- move validation to `MailingConfig` constructor
- perhaps `checkArgument` is better